src/chapters/control-flow/loops.rst

```diff
@@ -0,0 +1,278 @@
+integers in the array, it will stop after it finds the first matching
+instance. So once it finds the first ``10`` in the array, it prints "Found
+it!" and then terminates the loop. If the ``break`` statement weren’t
+there, the loop would continue and when it found the second ``10``, it
+would print "Found it!" a second time.
+
+Note that the ``break`` statement terminates the innermost loop that it
+is contained within. So if you have nested loops and use a ``break``
+statement within the innermost loop, then it will only terminate that
+loop and not the outer one.
+
+.. index:: ! continue
+
+Continue Statements in Loops
+----------------------------
+
+The **continue** statement is similar to, but importantly different
+from, the ``break`` statement. Like ``break``, it interrupts the normal
+flow of control of the loop. But unlike ``break``, the ``continue``
+statement only terminates the *current iteration* of the loop. So the
+loop will continue to run from the top (as long as the boolean
+expression that controls the loop is still true) after a ``continue``
+statement. Here is an example:
+
+.. sourcecode:: java
+   :linenos:
+
+   public class testContinue {
+
+      public static void main(String [] args) {
+         int[] someInts = {1, 10, 2, 3, 5, 8, 10};
+         int searchTerm = 10;
+         for (int oneInt : someInts) {
+            if (oneInt == searchTerm) {
+               System.out.println("Found it!");
+               continue;
+            }
+            System.out.println("Not here");
+         }
+      }
+   }
+
+The above program will print "Not here" on every iteration of the
+``for`` loop *except* where the number has been found. So the output
+looks like this:
+
+.. sourcecode:: bash
+
+   Not here
+   Found it!
+   Not here
+   Not here
+   Not here
+   Not here
+   Found it!
+
+Because of the ``continue`` statement, the final print statement in the
+for loop is skipped. If the ``continue`` statement weren’t there, the
+output would look like this instead (notice the extra "Not here"
+printouts):
+
+.. sourcecode:: bash
+
+   Not here
+   Found it!
+   Not here
+   Not here
+   Not here
+   Not here
+   Not here
+   Found it!
+   Not here
+
+References
+----------
+
+-  `The for statement
+   (docs.oracle.com) <https://docs.oracle.com/javase/tutorial/java/nutsandbolts/for.html>`__
+-  `The while and do-while Statements
+   (docs.oracle.com) <https://docs.oracle.com/javase/tutorial/java/nutsandbolts/while.html>`__
+-  `Break and Continue Statements
+   (docs.oracle.com) <https://docs.oracle.com/javase/tutorial/java/nutsandbolts/branch.html>`__
+-  `Summary of Control Flow Statements
+   (docs.oracle.com) <https://docs.oracle.com/javase/tutorial/java/nutsandbolts/flowsummary.html>`__
+
+Check Your Understanding
+-------------------------
+
+.. admonition:: Question
+
+   .. sourcecode:: java
+      :linenos:
+
+      char chars[] = {'p', 'l', 'r', 's', 't'};
+
+      for (<conditional>) {
```

Comment:
I think of a "conditional" as an ``if`` statement or boolean expression, but that's not what goes into a ``for each`` loop.  How about replacing ``<conditional>`` with some other placeholder?